### ISSUE TITLE
Remove app from `_app_registry` on app.stop()

### ISSUE THREAD

**Issue by cnicodeme** [sanic-org/sanic#2494]
**Describe the bug**
The `_app_registry` keeps hold of any app created by Sanic, even those stopped.

**Code snippet**
Adding the following code resolve the issue:

```
    def stop(self):
        """
        This kills the Sanic
        """
        if self.state.stage is not ServerStage.STOPPED:
            self.shutdown_tasks(timeout=0)
            for task in all_tasks():
                with suppress(AttributeError):
                    if task.get_name() == "RunServer":
                        task.cancel()
            get_event_loop().stop()

            # Added part:
            del self.__class__._app_registry[self.name]
```

**Additional context**
Implementing some advanced auto reload system causes issues when fully restarting an application.

(Restarting only the server (calling `app.run()`) won't take into consideration any changes such as the environment variables.)

**Comment by ahopkins** [comment]
A bit of scope creep in #2499, but this is added there for convenience since that PR is messing with the operation already.

**Comment by cnicodeme** [comment]
TLDR: I hope there will be an easy way to trigger a restart at Sanic from the CLI and outside the app.

---

I read the #2499 carefully. It seems exciting to have this new way of handling multiple processes.

One key issue I had while implementing my auto-reload script is to restart Sanic from an external script.

My use case is to deploy any new version pushed to "prod" on Github, so I've setup a webhook and a small Sanic server that listen to the endpoint specified at Github.
When receiving the request and ensuring it's for "prod", I call a "git pull" on my server's folder, followed by alembic upgrade, then, I restart Sanic.

Initially, I was relying on the auto-reload feature of Sanic to trigger a restart on file change, but if I had some DB modifications, the Alembic command wasn't run yet when Sanic would restart, resulting in errors and the app failed to restart.

That's why I had to control when the restart would occur, and I had to do that from an external script.

All to say, I hope there will be an easy way to trigger a restart at Sanic from the CLI and outside the app.

**Comment by cnicodeme** [comment]
Yes! Both are great!

I can either keep my external script running and call the CLI to restart the APP, or integrate the webhook directly in my app and restart it from itself.

Both have their advantages (the external script won't be impacted by a potential crash) and gives flexibility to the developer.

That's great!

(I had previously shared my script at https://community.sanicframework.org/t/auto-restarting-sanic-server-on-touch-of-a-specific-file/1039/2?u=cx42net). I'll post a new one once v22.9 will be released :)
